src/main/kotlin/io/github/nomisrev/service/ArticleService.kt

```diff
@@ -21,12 +23,17 @@ data class CreateArticle(
 interface ArticleService {
   /** Creates a new article and returns the resulting Article */
   suspend fun createArticle(input: CreateArticle): Either<DomainError, Article>
+
+  /** Get article by Slug */
+  suspend fun getArticleBySlug(slug: Slug): Either<DomainError, Article>
```

Comment:
```suggestion
  suspend fun getArticleBySlug(slug: Slug): Either<ArticleBySlugNotFound, Article>
```

Reply:
In this case, I think the error type has to be `DomainError `, because the method's implementation uses https://github.com/nomisRev/ktor-arrow-example/blob/main/src/main/kotlin/io/github/nomisrev/repo/UserPersistence.kt#L35 which already returns `Either<DomainError, UserInfo>`

Reply:
Aha, you're right! I missed that during my review. My apologies. Perfect PR 🙌 👏